client/asset/eth/nodeclient_harness_test.go

```diff
@@ -694,127 +687,56 @@ func TestInitiateBatch(t *testing.T) {
 		if err != nil {
 			t.Fatalf("unexpected error for test %v: %v", test.name, err)
 		}
-		txFee := big.NewInt(0).Mul(big.NewInt(int64(receipt.GasUsed)), gasPrice)
+
+		txFee := big.NewInt(0).Mul(big.NewInt(int64(receipt.GasUsed)), gweiToWei(baseFee))
 		wantBal := big.NewInt(0).Sub(originalBal, txFee)
 		if test.success {
-			wantBal.Sub(wantBal, test.txValue)
+			wantBal.Sub(wantBal, bigUint(test.txValue))
 		}
-		if bal.Cmp(wantBal) != 0 {
+		diff := big.NewInt(0).Sub(wantBal, bal)
+		if diff.CmpAbs(dexeth.BigGweiFactor) > 0 {
 			t.Fatalf("unexpected balance change for test %v: want %v got %v", test.name, wantBal, bal)
 		}
 
+		fmt.Printf("--comparing \n\n")
+
 		for _, testSwap := range test.swaps {
-			swap, err := ethClient.swap(ctx, simnetAcct, testSwap.SecretHash)
+
+			fmt.Printf("--checking swap %s\n", hex.EncodeToString(testSwap.SecretHash))
+
+			swap, err := ethClient.swap(ctx, bytesToArray(testSwap.SecretHash), 0)
 			if err != nil {
 				t.Fatalf("unexpected error for test %v: %v", test.name, err)
 			}
-			state := srveth.SwapState(swap.State)
-			if test.success && state != srveth.SSInitiated {
-				t.Fatalf("unexpected swap state for test %v: want %s got %s", test.name, srveth.SSInitiated, state)
+
+			fmt.Printf("--done checking swap %s \n\n", hex.EncodeToString(testSwap.SecretHash))
+
+			state := dexeth.SwapStep(swap.State)
+			if test.success && state != dexeth.SSInitiated {
+				t.Fatalf("unexpected swap state for test %v: want %s got %s", test.name, dexeth.SSInitiated, state)
```

Comment:
I cannot figure out why this isn't passing. I'm verifying the secret hashes match thoughout. There are no errors. I can see the transaction with `eth.getTransaction` afterwards, and I can see the secret hash in the tx data, but the `ETHSwapSwap` is completely zeroed.

Reply:
I've been trying to figure it out what's wrong.. haven't had any luck yet. 

I also saw that you removed the `addSignerToOpts` calls from the functions that submit transactions (`initiate`, `redeem`, etc.) and you're calling them from the test code, but in some cases you forgot to call it. What's wrong with this going into `intiate` and `redeem`?

Reply:
Here's a clue. 

https://github.com/decred/dcrdex/compare/master...buck54321:fail-init-real-balance?expand=1

That fails too.

Reply:
It's failing for any value > 1 gwei

Reply:
Update: I've got everything passing except for `TestReplayAttack`.

Reply:
Nice, what was wrong? I was getting a headache from that.

Reply:
I think I was handling a `*big.Int` wrong. Not making a copy when I should have or something.